src/Bicep.Cli/Services/CompilationService.cs

```diff
@@ -104,8 +109,27 @@ public ParamsSemanticModel CompileParams(string inputPath, bool skipRestore)
             {
                 throw new Exception($"Unable to read file {inputPath}");
             }
+            
+            var paramsFile = SourceFileFactory.CreateBicepParamFile(inputUri, fileText);
+
+            var usingDeclarations = paramsFile.ProgramSyntax.Children.OfType<UsingDeclarationSyntax>();
+
+            if(!PathHelper.TryGetUsingPath(usingDeclarations.SingleOrDefault(), out var bicepFilePath, out var failureBuilder))
```

Comment:
If you throw an exception, the user experience will be that they only see the one diagnostic for the exception being thrown - any other validation will be lost. To the user they will see unrelated squigglies appearing and disappearing while they are typing. 

As a general rule, we try to validate **as much** as possible in an open document - meaning you're going to be better off raising a diagnostic and allowing the rest of the validation to continue rather than throwing an exception.